camel-core/src/main/java/org/apache/camel/Exchange.java

```diff
@@ -169,6 +169,8 @@
     String MAXIMUM_TRANSFORMER_CACHE_SIZE = "CamelMaximumTransformerCacheSize";
     String MAXIMUM_VALIDATOR_CACHE_SIZE = "CamelMaximumValidatorCacheSize";
     String MESSAGE_HISTORY             = "CamelMessageHistory";
+    String MESSAGE_HISTORY_HEADER      = "CamelMessageHistoryHeader";
+    String MESSAGE_HISTORY_OUTPUT      = "CamelMessageHistoryOutput";
```

Comment:
I'm not entirely sure that we should use that wording, could `String MESSAGE_HISTORY_HEADER_FORMAT = "CamelMessageHistoryHeaderFormat"` and `String MESSAGE_HISTORY_OUTPUT_FORMAT = "CamelMessageHistoryOutputFormat"` be better, so we don't mix them with message headers?